features/step_definitions/carousel_steps.rb

```diff
@@ -27,6 +27,15 @@
   end
 end
 
+Given /I create four new products.*/ do
+  step %{Given a vendor already exists}
+  FactoryBot.create(:product, name: "First Product")
+  FactoryBot.create(:product, name: "Second Product")
+  FactoryBot.create(:product, name: "Third Product")
+  FactoryBot.create(:product, name: "Four Product")
+end
+
+
```

Comment:
The above step def is not needed. There is a step def in index_steps.rb that does this for a dynamic number of vendors/products. "Given/there are (.*)(.*)/" will allow us to create any number of products or vendors. We still need the "step %{Given a vendor already exists}" though.